include/bm/bm_sim/source_info.h

```diff
@@ -0,0 +1,85 @@
+/* Copyright 2017 Cisco Systems, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*
+ * Andy Fingerhut (jafinger@cisco.com)
+ *
+ */
+
+#ifndef BM_BM_SIM_SOURCE_INFO_H_
+#define BM_BM_SIM_SOURCE_INFO_H_
+
+//! @file source_info.h
+//! Stores source location information about some objects, as read from
+//! the JSON file produced by the compiler.
+
+#include <string>
+#include "jsoncpp/json.h"
```

Comment:
It would be easy to avoid introducing this dependency in the header file. You could forward declare `Json::Value` and move the implementation of `newFromJson` to an implementation file. Or better still, I think the json parsing code should stay in `P4Objects.cpp`; it would be easier to do sanity checking on the json if you left it there (because of the code already defined there), and would keep the json parsing in a single place.